File: Problem015.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package projecteuler;

/**
 *
 * @author Manan
 */
public class Problem015 extends Library {
    public void euler015()
    {
        // n choose k algorithm 
        // for any grid, n is the number of rows* number of columns
        // and k is the number of rows
        int n = 4;
        int k = 2;

        if (k > (n - k)) 
            k = n - k;

        long c = 1;
        for (int i = 0; i < k; i++) {
            c *= n - i;
            c /= i + 1;
        }
        System.out.println(c);
    }
}

File: euler/Problem015.cpp
#include "library.h";
using namespace std;

ll p015()
{
	// n choose k algorithm 
	// for any grid, n is the number of rows + number of columns
	// and k is the number of rows
	//same implementation in java and haskell

	ll n = 40;
	ll k = 20;

	if (k > (n - k))
		k = n - k;

	ll c = 1;
	for (ll i = 0; i < k; i++) {
		c *= n - i;
		c /= i + 1;
	}
	return c;
}